epam_tr/src/main/java/com/epam/tr/controller/UserController.java

```diff
@@ -0,0 +1,53 @@
+package com.epam.tr.controller;
+
+import com.epam.tr.entities.AppUser;
+import com.epam.tr.exceptions.InvalidCredentialsException;
+import com.epam.tr.service.UserServiceImpl;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.web.bind.annotation.*;
+
+import static org.springframework.http.HttpStatus.*;
+
+@RestController
+@RequestMapping("/users")
+public class UserController {
+
+    @Autowired
+    private UserServiceImpl service;
+    @Autowired
+    private BCryptPasswordEncoder encoder;
+
```

Comment:
READ REST API STANDARDS.

Reply:
and this article - https://hackernoon.com/restful-api-designing-guidelines-the-best-practices-60e1d954e7c9